### DISCUSSION TITLE
Compare multiple commits

### DISCUSSION THREAD

**Discussion by marekzan** [jesseduffield/lazygit#2142] [Q&A] [answered]
Hey,

so I love the feature of the intellij IDEA git functionality, that you can mark multiple commits (more than two) and compare them. So I would be able to see every new change in one sitting instead of going through multiple commits and maybe encounter changes wich become obsolete in the remaining commits.

Is this kind of functionality possible with lazygit? I read that you could compare two commits only?

**Comment by mark2185** [comment, marked as answer]
How would that work from the `cli`?

I don't use `IDEA` and I'm having a hard time imagining what would e.g. 4 commits be compared to? I can see in `man git diff` that there is such a feature for `merge` commits, but I'd rather hear it from you.

One could position the cursor on a commit in the `Commits` panel and then press `W` to `open diff menu` where one can compare the currently selected point in the history with a different branch/commit. Would that maybe help?

**Comment by marekzan** [comment]
Ah yes! The `W` was the answer. So in IDEA you could mark multiple commits and see all changes made in these commits at once, and the `open diff menu` is doing exactly that. I thought it would really only compare two commits and not `from` this commit `to` this commit.
Sorry for taking your time, and thank you again very much for your answer!
